server/hotrod/src/main/scala/org/infinispan/server/hotrod/iteration/IterationManager.scala

```diff
@@ -98,7 +98,7 @@ class DefaultIterationManager(val cacheManager: EmbeddedCacheManager) extends It
       })
 
       val segmentListener = new IterationSegmentsListener
-      val compatInfo = CompatInfo(cacheManager.getCacheConfiguration(cacheName).compatibility())
+      val compatInfo = CompatInfo(cacheManager.getCache(cacheName).getCacheConfiguration.compatibility())
```

Comment:
For some reason ```cacheManager.getCacheConfiguration(cacheName).compatibility()``` returns the wrong cache configuration (from the default one) instead of the ```cacheName```'s cache configuration. 
This needs more investigation (another JIRA)